src/backend/distributed/worker/worker_shard_visibility.c

```diff
@@ -334,6 +336,26 @@ ResetHideShardsDecision(void)
 static bool
 ShouldHideShardsInternal(void)
 {
+	if (MyBackendType == B_BG_WORKER &&
+		IsPgBgWorker())
+	{
+		/*
+		 * If a background worker belongs to Postgres, we should
+		 * never hide shards. For other background workers, enforce
+		 * the application_name check.
+		 */
+		return false;
+	}
+	else if (MyBackendType != B_BACKEND)
```

Comment:
this still filters returns false for pg_cron background workers I think

Reply:
Oh, aren't they `MyBackendType == B_BG_WORKER`? Let me try

Reply:
As far as I can debug, the bg workers that are runniing the cron jobs are also bg workers:
```SQL
│ pid              │ 45585                                                                                                                                                                                                                                    │
│ query            │ SELECT pg_sleep(100)                                                                                                                                                                                                                     │
│ application_name │                                                                                                                                                                                                                                          │
│ backend_type     │ pg_cron                                                                                                                                         
```

```bash
p MyBackendType
(BackendType) $1 = B_BG_WORKER
```


Note that we also have the following, which is also not filtered

```
│ backend_type     │ pg_cron launcher                                                                                                                                                                                                                         
```